pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -484,42 +485,42 @@ func (r *PerformanceProfileReconciler) Reconcile(ctx context.Context, req ctrl.R
 	if err != nil {
 		klog.Errorf("failed to deploy performance profile %q components: %v", instance.GetName(), err)
 		r.Recorder.Eventf(instance, corev1.EventTypeWarning, "Creation failed", "Failed to create all components: %v", err)
-		conditions := getDegradedConditions(conditionReasonComponentsCreationFailed, err.Error())
-		if err := r.updateStatus(instance, conditions); err != nil {
+		conditions := status.GetDegradedConditions(status.ConditionReasonComponentsCreationFailed, err.Error())
+		if err := status.UpdateStatus(ctx, r.Client, instance, conditions); err != nil {
 			klog.Errorf("failed to update performance profile %q status: %v", instance.GetName(), err)
 			return reconcile.Result{}, err
 		}
 		return reconcile.Result{}, err
 	}
 
 	// get kubelet false condition
-	conditions, err := r.getKubeletConditionsByProfile(instance.GetName())
+	conditions, err := status.GetKubeletConditionsByProfile(ctx, r.Client, instance.GetName())
 	if err != nil {
-		return r.updateDegradedCondition(instance, conditionFailedGettingKubeletStatus, err)
+		return r.updateDegradedCondition(instance, status.ConditionFailedGettingKubeletStatus, err)
 	}
 
 	// get MCP degraded conditions
 	if conditions == nil {
-		conditions, err = getMCPDegradedCondition(profileMCP)
+		conditions, err = status.GetMCPDegradedCondition(profileMCP)
 		if err != nil {
-			return r.updateDegradedCondition(instance, conditionFailedGettingMCPStatus, err)
+			return r.updateDegradedCondition(instance, status.ConditionFailedGettingMCPStatus, err)
 		}
 	}
 
 	// get tuned profile degraded conditions
 	if conditions == nil {
-		conditions, err = r.getTunedConditionsByProfile(instance)
+		conditions, err = status.GetTunedConditionsByProfile(ctx, r.Client, instance)
 		if err != nil {
-			return r.updateDegradedCondition(instance, conditionFailedGettingTunedProfileStatus, err)
+			return r.updateDegradedCondition(instance, status.ConditionFailedGettingTunedProfileStatus, err)
 		}
 	}
 
 	// if conditions were not added due to machine config pool status change then set as available
 	if conditions == nil {
-		conditions = getAvailableConditions("")
+		conditions = status.GetAvailableConditions("")
 	}
 
-	if err := r.updateStatus(instance, conditions); err != nil {
+	if err := status.UpdateStatus(ctx, r.Client, instance, conditions); err != nil {
```

Comment:
nit: `status.Update` is a little nicer, if we can do this rename (not sure there are other implications e.g. 1000 usages)

Reply:
Done